### PR TITLE
feat(cdk): bidirectional communication via gRPC

### PR DESCRIPTION
**Pull request by afiune**
## Introducing Remote Procedure Calls (gRPC)

There are circumstances in which a component may need to communicate back to the Lacework CLI, for this communication (Component ← → CDK) the CDK architecture provides a gRPC server that running components can connect to using the target exported via the environment variable `LW_CDK_TARGET`.

We have chosen gRPC since it provides extreme flexibility to run on any environment and it has support on the most popular programming languages.

A quick snipped of Go code that shows how to `ping` the CDK server:

```go
var log = lwlogger.New(os.Getenv("LW_LOG")).Sugar()

// Set up a connection to the CDK server
log.Infow("connecting to gRPC server", "address", os.Getenv("LW_CDK_TARGET"))
conn, err := grpc.Dial(,
	// we allow insecure connections since we are connecting to 'localhost'
	grpc.WithTransportCredentials(insecure.NewCredentials()))
if err != nil {
	return errors.Wrap(err, "could not connect")
}
defer conn.Close()

var (
	cdkClient   = cdk.NewCoreClient(conn)
	ctx, cancel = context.WithTimeout(context.Background(), time.Second)
)
defer cancel()

// Ping the CDK server
reply, err := cdkClient.Ping(ctx, &cdk.PingRequest{
	ComponentName: os.Getenv("LW_COMPONENT_NAME"),
})
if err != nil {
	return errors.Wrap(err, "could not ping")
}
log.Debugw("response", "from", "cdk.v1.Core/Ping", "message", reply.GetMessage())
```

Docs: https://lacework.atlassian.net/l/cp/3y1TAREj

Signed-off-by: Salim Afiune Maya <afiune@lacework.net>

## How did you test this change?

I used my `lookup` component. https://github.com/afiune/lookup/pull/1

Installed the component ☝🏽 and ran a few commands:
```
$ lacework lookup user:root --debug
{"level":"info","ts":"2022-11-16T13:57:40-08:00","caller":"cmd/cdk.go:110","msg":"gRPC server started","address":"127.0.0.1:1123"}
{"level":"info","ts":"2022-11-16T13:57:40-08:00","caller":"lookup/main.go:36","msg":"connecting to gRPC server","address":"localhost:1123"}
{"level":"debug","ts":"2022-11-16T13:57:41-08:00","caller":"cmd/cdk.go:70","msg":"message","from":"CDK/Honeyvent","feature":"lookup_event"}
{"level":"debug","ts":"2022-11-16T13:57:41-08:00","caller":"cmd/honeyvent.go:191","msg":"new honeyvent","dataset":"lacework-cli-dev","trace_id":"3d6f908dde1af75a","span_id":"4bf8eb96aff735a8","parent_id":"2856631e20e1047c"}
{"level":"debug","ts":"2022-11-16T13:57:41-08:00","caller":"cmd/honeyvent.go:203","msg":"sending honeyvent","dataset":"lacework-cli-dev"}
```

### We see events! 💯 

Screen Shot 2022-11-16 at 1 56 19 PM

Plus tests. 🙌🏽 

## Issue

https://lacework.atlassian.net/browse/ALLY-1026

### CODE REVIEW

File: cli/cdk/cdk_grpc.pb.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
+// Code generated by protoc-gen-go-grpc. DO NOT EDIT.
```
Comment (dmurray-lacework) [inline cli/cdk/cdk_grpc.pb.go:1]: Should we move autogen code to a subfolder? 

cli/cdk/build/cdk_grpc.pb.go ?
cli/cdk/generated/cdk_grpc.pb.go ?

Comment (afiune) [reply cli/cdk/cdk_grpc.pb.go:1]: any reason why?

Comment (dmurray-lacework) [reply cli/cdk/cdk_grpc.pb.go:1]: It's clear from a glance at a project what is autogen and what is not. Take this PR for example, without having to click into the file and read the comment I would know to ignore this file. This is fine for 1 file in a PR, but what about > 10 autogen files living among code that we should be reviewing.

Comment (afiune) [reply cli/cdk/cdk_grpc.pb.go]: moved them 👍🏽

**Comment by afiune** [comment]
check windows

**Comment by afiune** [comment]
check windows

**Comment by afiune** [comment]
check windows 🪟 pleeeeeease

**Review by jon-stewart — APPROVED** [review]
Interesting to have POC grpc server launched with component command.  Hidden development code and can easily be ripped out in future /shrug.

Neat.

**Comment by afiune** [comment]
I upgraded all packages I could upgrade but we still have vulnerabilities in `main`. I am going to merge this and work on fixing them on a different PR.

tenor-106878847

**Pull request merged**